### ISSUE TITLE
Configuration injection isn't supported for individual constructor properties

### ISSUE THREAD

**Issue by distinctdan** [micronaut-projects/micronaut-core#11333]
### Expected Behavior

I need to inject several configuration properties into my singleton's constructor. However, the properties are in different locations in the configuration and Micronaut only seems to support binding all parameters of a constructor at the same time to a single configuration object via `@ConfigurationInject`. This isn't flexible and doesn't work for my use case, this should really support @Value to be able to inject individual properties. But, the docs say that @Value explicityly isn't supported. Since it's possible to inject configuration into a constructor, is there any reason not to support individual properties?

Here's my desired API, I would also be ok with using a variant of `@Value` like maybe `@ConfigValue` if it needs to be different:
```kotlin
@Singleton
class MyTokenValidator<R> @Inject constructor(
    @Value("\${micronaut.oktaBaseUrl}")
    val expectedIssuer: String,
    
    @Value("\${micronaut.security.token.jwt.claims-validators.keys}")
    val jwksKeysUrl: String,
) : TokenValidator<R> {
```

### Actual Behaviour

Injecting individual constructor properties isn't supported.

### Version

4.6.3

**Comment by glorrian** [comment]
Micronaut supports constructor-injection with @Value. You can run this test and it will pass
```groovy
package com.example

import io.micronaut.context.annotation.Value
import io.micronaut.test.extensions.spock.annotation.MicronautTest
import jakarta.inject.Inject
import jakarta.inject.Singleton
import spock.lang.Specification

@MicronautTest(propertySources = "classpath:application-test.properties")
class ValueTest extends Specification {

    @Inject
    TestService testService

    void "test value annotation into constructor"() {
        expect:
        testService.myConfigProperty == "foo"
        testService.otherConfigProperty == "bar"
    }

    @Singleton
    static class TestService {
        String myConfigProperty
        String otherConfigProperty

        TestService(
                @Value('${my.config.property}') String myConfigProperty,
                @Value('${other.config.property}') String otherConfigProperty
        ) {
            this.myConfigProperty = myConfigProperty
            this.otherConfigProperty = otherConfigProperty
        }
    }

}
```
application-test.properties:
```properties
my.config.property=foo
other.config.property=bar
```

**Comment by graemerocher** [comment]
you can also use `@Property(name="my.config.property")` of you don't want a value expression. I am not sure what more we can do here

**Comment by distinctdan** [comment]
Thank you guys for the examples! After more debugging, I figured it out that I needed a separate yml dependency for v4 in order to parse the configuration file. At some point, I had switched it from a .properties to a .yml. Then, when I was reading the docs, there isn't an example of injecting a property or value in a constructor, and I saw a note under `@ConfigurationProperties` that said it didn't work for `@Value`, which is what led me to believe that constructor injection wasn't supported. Thanks for the help getting this figured out.
